### PR TITLE
When pipeline run didn't succeeded and also didn't finish, cancel it

### PR DESCRIPTION
**Pull request by rcerven**
* CLOUDBLD-8220

Signed-off-by: Robert Cerven <rcerven@redhat.com>

# Maintainers will complete the following section

- [ ] Commit messages are descriptive enough
- [ ] Code coverage from testing does not decrease and new code is covered
- [ ] JSON/YAML configuration changes are updated in the relevant schema
- [ ] Pull request has a link to an osbs-docs PR for user documentation updates

### CODE REVIEW

File: koji_containerbuild/plugins/builder_containerbuild.py [hunk 1]
```diff
@@ -537,6 +537,13 @@ def sigint_handler(*args, **kwargs):
 
         elif not has_succeeded:
             error_message = self.osbs().get_build_error_message(build_id)
+
+            if self.osbs().build_not_finished(build_id):
+                try:
+                    self.osbs().cancel_build(build_id)
+                except Exception:
+                    pass
```
Comment (mkosiarc) [inline koji_containerbuild/plugins/builder_containerbuild.py:545]: Should this be empty?

Comment (rcerven) [reply koji_containerbuild/plugins/builder_containerbuild.py:545]: well we won't see logs from koji-c anyway so it doesn't matter at all

Comment (ben-alkov) [reply koji_containerbuild/plugins/builder_containerbuild.py:545]: So if canceling the build raises an exception, we just ignore it and continue?
It looks to me like the exception is "handled" by falling through to `if error_message`. Do we have any guarantee that `get_build_error_message` will *never* come up empty...?

Comment (rcerven) [reply koji_containerbuild/plugins/builder_containerbuild.py:545]: 1st I am getting original error message

then IF pipeline run isn't finished I cancel pipeline run (and I don't care anymore about it, I want to propagate only the original message, not that "additional" cancel failed)

Comment (rcerven) [reply koji_containerbuild/plugins/builder_containerbuild.py:545]: if error_message is empty that is handled bellow, but how is that relevant?

Comment (ben-alkov) [reply koji_containerbuild/plugins/builder_containerbuild.py:545]: I'm concerned about what happens if canceling the build fails with an exception

Comment (rcerven) [reply koji_containerbuild/plugins/builder_containerbuild.py:545]: build fails with the original error

Comment (chmeliik) [reply koji_containerbuild/plugins/builder_containerbuild.py:545]: Could we log the exception anyway? Maybe we won't see it, but just out of principle.

Comment (rcerven) [reply koji_containerbuild/plugins/builder_containerbuild.py:545]: sure

**Review by chmeliik — APPROVED** [review]
LGTM :+1:

**Review by ben-alkov — APPROVED** [review]

**Comment by rcerven** [comment]
no rn, fix for unreleased feature

**Pull request merged**